Fix project data on home and project pages

terceiro opened the pull request.

mwasilew commented on squad/core/models.py:
```diff
@@ -374,7 +358,28 @@ def overall(self):
+
+    @property
+    def has_tests(self):
+        return (self.tests_pass + self.tests_fail) > 0
```
hmm, maybe 'missing' should also be included here? In general I don't think that 'missing' is a proper name. 'skip' is a valid result and we should support it at some point.

terceiro commented:
@mwasilew rebased against current master and addressed your comment wrt `missing` (replaced it with `skip`), and missing will be what we currently call `n/a`, i.e. results that are not there.

mwasilew commented:
I got the following exception trying to apply migrations:

```
  Applying core.0034_prepare_to_remove_build_name...Traceback (most recent call last):
  File "/home/milosz/linaro/tcstorm/squad_venv/lib/python3.5/site-packages/django/db/backends/utils.py", line 63, in execute
    return self.cursor.execute(sql)
  File "/home/milosz/linaro/tcstorm/squad_venv/lib/python3.5/site-packages/django/db/backends/sqlite3/base.py", line 326, in execute
    return Database.Cursor.execute(self, query)
sqlite3.IntegrityError: UNIQUE constraint failed: core_build.project_id, core_build.version

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "./manage.py", line 5, in <module>
    main()
  File "/home/milosz/linaro/tcstorm/squad/squad/manage.py", line 11, in main
    execute_from_command_line(sys.argv)
  File "/home/milosz/linaro/tcstorm/squad_venv/lib/python3.5/site-packages/django/core/management/__init__.py", line 363, in execute_from_command_line
    utility.execute()
  File "/home/milosz/linaro/tcstorm/squad_venv/lib/python3.5/site-packages/django/core/management/__init__.py", line 355, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/home/milosz/linaro/tcstorm/squad_venv/lib/python3.5/site-packages/django/core/management/base.py", line 283, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/home/milosz/linaro/tcstorm/squad_venv/lib/python3.5/site-packages/django/core/management/base.py", line 330, in execute
    output = self.handle(*args, **options)
  File "/home/milosz/linaro/tcstorm/squad_venv/lib/python3.5/site-packages/django/core/management/commands/migrate.py", line 204, in handle
    fake_initial=fake_initial,
  File "/home/milosz/linaro/tcstorm/squad_venv/lib/python3.5/site-packages/django/db/migrations/executor.py", line 115, in migrate
    state = self._migrate_all_forwards(state, plan, full_plan, fake=fake, fake_initial=fake_initial)
  File "/home/milosz/linaro/tcstorm/squad_venv/lib/python3.5/site-packages/django/db/migrations/executor.py", line 145, in _migrate_all_forwards
    state = self.apply_migration(state, migration, fake=fake, fake_initial=fake_initial)
  File "/home/milosz/linaro/tcstorm/squad_venv/lib/python3.5/site-packages/django/db/migrations/executor.py", line 244, in apply_migration
    state = migration.apply(state, schema_editor)
  File "/home/milosz/linaro/tcstorm/squad_venv/lib/python3.5/site-packages/django/db/migrations/migration.py", line 129, in apply
    operation.database_forwards(self.app_label, schema_editor, old_state, project_state)
  File "/home/milosz/linaro/tcstorm/squad_venv/lib/python3.5/site-packages/django/db/migrations/operations/special.py", line 108, in database_forwards
    self._run_sql(schema_editor, self.sql)
  File "/home/milosz/linaro/tcstorm/squad_venv/lib/python3.5/site-packages/django/db/migrations/operations/special.py", line 133, in _run_sql
    schema_editor.execute(statement, params=None)
  File "/home/milosz/linaro/tcstorm/squad_venv/lib/python3.5/site-packages/django/db/backends/base/schema.py", line 120, in execute
    cursor.execute(sql, params)
  File "/home/milosz/linaro/tcstorm/squad_venv/lib/python3.5/site-packages/django/db/backends/utils.py", line 80, in execute
    return super(CursorDebugWrapper, self).execute(sql, params)
  File "/home/milosz/linaro/tcstorm/squad_venv/lib/python3.5/site-packages/django/db/backends/utils.py", line 65, in execute
    return self.cursor.execute(sql, params)
  File "/home/milosz/linaro/tcstorm/squad_venv/lib/python3.5/site-packages/django/db/utils.py", line 94, in __exit__
    six.reraise(dj_exc_type, dj_exc_value, traceback)
  File "/home/milosz/linaro/tcstorm/squad_venv/lib/python3.5/site-packages/django/utils/six.py", line 685, in reraise
    raise value.with_traceback(tb)
  File "/home/milosz/linaro/tcstorm/squad_venv/lib/python3.5/site-packages/django/db/backends/utils.py", line 63, in execute
    return self.cursor.execute(sql)
  File "/home/milosz/linaro/tcstorm/squad_venv/lib/python3.5/site-packages/django/db/backends/sqlite3/base.py", line 326, in execute
    return Database.Cursor.execute(self, query)
django.db.utils.IntegrityError: UNIQUE constraint failed: core_build.project_id, core_build.version
```

mwasilew commented:
OK, I downloaded fresh production db and the problem is gone.
